static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx

```diff
@@ -18,60 +22,113 @@ export function transformMetricsToArea<T extends WidgetDataConstraint>(
 
   const {errored, loading, reloading, response, responsePrevious} = results;
 
+  const commonChildData = {
+    loading,
+    reloading,
+    isLoading: loading || reloading,
+    isErrored: errored,
+
+    utc: utc === 'true',
+    interval,
+    statsPeriod: statsPeriod ?? undefined,
+    start: start ?? '',
+    end: end ?? '',
+  };
+
+  if (!response || !responsePrevious) {
+    return {
+      ...commonChildData,
+      hasData: false,
+      data: [],
+      dataMean: undefined,
+      previousData: undefined,
+    };
+  }
+
   const metricsField = widgetProps.fields[0];
 
-  const data =
-    response?.groups.map(group => {
-      return {
-        seriesName: metricsField,
-        data: response.intervals.map((intervalValue, intervalIndex) => {
-          return {
-            name: moment(intervalValue).valueOf(),
-            value: group.series[metricsField][intervalIndex],
-          };
-        }),
-      };
-    }) ?? [];
-
-  const dataMean = data.map(series => {
-    const meanData = mean(series.data.map(({value}) => value));
+  const isFailureRateWidget =
```

Comment:
Hopefully, this whole transform function will be simplified once there's a failure_rate field in API.